### PR TITLE
[cxx-interop] Add benchmark for std::set conversion to a Swift collection

### PR DESCRIPTION
**Pull request by egorzhdan**
This adds a benchmark for the C++ stdlib overlay functionality that converts C++ containers to Swift collections.

This also fixes a Python exception in the benchmarking engine that happened when no memory measurements were taken for a particular benchmark

### CODE REVIEW

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please benchmark

**Comment by hyp** [comment]
what exactly are you benchmarking here, the cast from set to Array, or the Array iteration?

**Comment by egorzhdan** [comment]
@hyp the intention is to benchmark the Array initializer, I added a for-in loop to make sure that it's not removed by the optimizer. Do you know a better way to benchmark this?

**Comment by egorzhdan** [comment]
@swift-ci please benchmark

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please benchmark

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by hyp** [comment]
You should just put the constructed array value into the `blackHole`. The compiler will not remove it then.

**Comment by egorzhdan** [comment]
I haven't thought of this, thanks!

**Comment by egorzhdan** [comment]
@swift-ci please benchmark

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please benchmark

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please benchmark

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please benchmark

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please benchmark

**Comment by egorzhdan** [comment]
@swift-ci please benchmark

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please benchmark

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please benchmark

**Comment by egorzhdan** [comment]
@swift-ci please benchmark

**Comment by egorzhdan** [comment]
@swift-ci please benchmark

**Comment by egorzhdan** [comment]
@swift-ci please benchmark

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please benchmark

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

File: benchmark/scripts/Benchmark_Driver [hunk 1]
```diff
@@ -627,16 +627,26 @@ class BenchmarkDoctor(object):
     @staticmethod
     def _constant_memory_use(measurements):
         select = BenchmarkDoctor._select
+        name = measurements["name"]
+
+        memory_uses = [
+            [r.mem_pages for r in i_series if r.mem_pages is not None]
+            for i_series in [select(measurements, num_iters=i) for i in [1, 2]]
+        ]
+        memory_uses = [m for m in memory_uses if m]
```
Comment (egorzhdan) [inline benchmark/scripts/Benchmark_Driver:636]: @tbkka could you please take a look at this part of the PR? This fixes a `TypeError` that was thrown out of this Python code when `memory_uses` is either empty or only contains empty elements.

cc @shahmishal

Comment (tbkka) [reply benchmark/scripts/Benchmark_Driver:636]: Looks fine.

**Pull request merged**